torch/csrc/distributed/autograd/context/container.cpp

```diff
@@ -157,24 +157,38 @@ void DistAutogradContainer::releaseContext(int64_t context_id) {
 }
 
 void DistAutogradContainer::sendReleaseContextRpc(int64_t context_id) {
-  // notify other workers to clean up their contexts.
+  // Best-effort notification to other workers to clean up their Dist autograd
+  // context, in order to reduce memory usage.
   auto workerIds =
       autograd_context_.find(context_id)->second->getKnownWorkerIds();
   // agent.send() or getCurrentRpcAgent may throw an error in the case of an
   // ungraceful shutdown, where we are shutting down RPC and also processing
   // this message in a separate thread concurrently. In this case, don't throw
   // here.
+  std::shared_ptr<rpc::RpcAgent> agent = nullptr;
   try {
-    auto agent = rpc::RpcAgent::getCurrentRpcAgent();
-    for (const auto& worker_id : workerIds) {
+    agent = rpc::RpcAgent::getCurrentRpcAgent();
+  } catch (const std::exception& e) {
+    LOG(INFO)
+        << "Failed to send RPC to clear Dist Autograd context to all workers: "
+        << e.what();
+    return;
+  }
+
+  TORCH_INTERNAL_ASSERT(agent, "RPC Agent should be set.");
+
+  for (const auto& worker_id : workerIds) {
+    try {
+      // Try to send RPC to clear context, though continue attempting to send to
```

Comment:
though -> through? (not sure)

Reply:
I meant for it to be "though", to explain that we want to continue sending in case one fails. I think I can word this better though.